Enable hash converters behind feature

vinistock opened the pull request:
Closes #349.

This PR puts all the hash converters that were commented out in #348 behind a feature.

Please do provide feedback about the module and the feature name. Also, let me know if I missed something.

lopopolo commented on artichoke-backend/Cargo.toml:
```diff
@@ -52,3 +52,4 @@ artichoke-array = []
 artichoke-debug = ["backtrace"]
 artichoke-random = ["rand"]
 artichoke-system-environ = []
+artichoke-all-converter = []
```
can you alphabetize this list?

lopopolo replied:
based on the flag used in the code, it looks like this should be

```toml
artichoke-all-converters = []
```

converters with the trailing `s`.

lopopolo commented on artichoke-backend/src/convert/array.rs:
```diff
@@ -184,153 +185,304 @@ array_to_ruby!(&'a [Option<Float>]);
-// array_to_ruby!(HashMap<&'a str, String>);
-// array_to_ruby!(HashMap<&'a str, &'a str>);
+#[cfg(feature = "artichoke-all-converters")]
+mod hash_convertions {
```
can this module be called `optional`?

lopopolo commented on artichoke-backend/src/convert/array.rs:
```diff
@@ -1,4 +1,5 @@
-// use std::collections::HashMap;
+#[cfg(feature = "artichoke-all-converters")]
+use std::collections::HashMap;
```
if this `use` statement is only required by the optional converters which are nested in a module, can you add that use directive into that module?

As it stands, I don't know if this compiles.

vinistock commented:
I've addressed your comments. Completely missed the hash and nilable files, apologies!

I used `use super::*;` to get the parent module's imports and definitions. I'm unsure if that's the appropriate way of doing it. Please let me know if you have other concerns!
